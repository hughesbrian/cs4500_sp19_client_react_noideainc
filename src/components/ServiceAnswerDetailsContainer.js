import React from 'react'
import ServiceAnswerService from '../services/ServiceAnswerService'
import ServiceQuestionService from '../services/ServiceQuestionService'
import ServiceAnswerDetails from './ServiceAnswerDetails'
class ServiceAnswerDetailsContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceAnswerService = ServiceAnswerService.getInstance();
        this.serviceQuestionService = ServiceQuestionService.getInstance();
        this.state = {
            serviceAnswers: [],
            answer: {
                choiceAnswer: '',
                id: 0,
                trueFalseAnswer: '',
                minRangeAnswer: '',
                maxRangeAnswer: ''
            },
            createAnswer:{
                choiceAnswer: '',
                id: 0,
                trueFalseAnswer: '',
                minRangeAnswer: '',
                maxRangeAnswer: ''
            },

            serviceQuestions : [],
            serviceQuestion: {
                question: '',
                id: 1
            }
        }
    }
    componentDidMount() {
        this.findAllServiceAnswers();
        this.serviceQuestionService
            .findAllServiceQuestions()
            .then(serviceQuestions => {
            this.setState({
                serviceQuestions: serviceQuestions,
                serviceQuestion: serviceQuestions[0]
            })
        }
    )
    }
    findAllServiceAnswers = () =>
    this.serviceAnswerService
        .findAllServiceAnswers()
        .then(serviceAnswers => {
            //this.props.history.push("/admin/service-answers/" + serviceAnswers[0].id);
            this.setState({
                      serviceAnswers: serviceAnswers,
                      answer: serviceAnswers[0]
                  })
}
)
selectServiceAnswer = id =>
this.serviceAnswerService
    .findServiceAnswerById(id)
    .then(answer => {
    //this.props.history.push("/admin/service-answers/" + id);
    this.setState({
        answer: answer
    })
}
)

updateMinValue = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: e.target.value,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateChoiceAnswer = e =>
this.setState({
    answer: {
        choiceAnswer: e.target.value,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateMaxValue = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: this.state.answer.trueFalseAnswer,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: e.target.value,
        id: this.state.answer.id
    }
})

updateTrueFalse = e =>
this.setState({
    answer: {
        choiceAnswer: this.state.answer.choiceAnswer,
        trueFalseAnswer: e.target.value,
        minRangeAnswer: this.state.answer.minRangeAnswer,
        maxRangeAnswer: this.state.answer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewMinValue = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: e.target.value,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewChoiceAnswer = e =>
this.setState({
    createAnswer: {
        choiceAnswer: e.target.value,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})

updateNewMaxValue = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: this.state.createAnswer.trueFalseAnswer,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: e.target.value,
        id: this.state.answer.id
    }
})

updateNewTrueFalse = e =>
this.setState({
    createAnswer: {
        choiceAnswer: this.state.createAnswer.choiceAnswer,
        trueFalseAnswer: e.target.value,
        minRangeAnswer: this.state.createAnswer.minRangeAnswer,
        maxRangeAnswer: this.state.createAnswer.maxRangeAnswer,
        id: this.state.answer.id
    }
})



selectServiceQuestion = id =>
this.serviceQuestionService
    .findServiceQuestionById(id)
    .then(serviceQuestion => {
    this.setState({
        serviceQuestion: serviceQuestion
    })
}
)

createServiceAnswer = () =>
this.serviceAnswerService
    .createServiceAnswer(this.state.createAnswer)
    .then(this.findAllServiceAnswers());
deleteServiceAnswer = id =>
this.serviceAnswerService
    .deleteServiceAnswer(id)
    .then(this.findAllServiceAnswers());
updateServiceAnswer = () =>
this.serviceAnswerService.updateServiceAnswer(this.state.answer);

render = () => <ServiceAnswerDetails serviceAnswers = {this.state.serviceAnswers}
                answer = {this.state.answer}
                createAnswer = {this.state.createAnswer}
                serviceQuestion = {this.state.serviceQuestion}
                serviceQuestions = {this.state.serviceQuestions}
                findAllServiceAnswer = {this.findAllServiceAnswers}
                selectServiceAnswer = {this.selectServiceAnswer}
                updateMinValue = {this.updateMinValue}
                updateChoiceAnswer = {this.updateChoiceAnswer}
                updateMaxValue = {this.updateMaxValue}
                updateTrueFalse = {this.updateTrueFalse}
                updateNewMinValue = {this.updateNewMinValue}
                updateNewChoiceAnswer = {this.updateNewChoiceAnswer}
                updateNewMaxValue = {this.updateNewMaxValue}
                updateNewTrueFalse = {this.updateNewTrueFalse}
                selectServiceQuestion = {this.selectServiceQuestion}
                createServiceAnswer = {this.createServiceAnswer}
                deleteServiceAnswer = {this.deleteServiceAnswer}
                updateServiceAnswer = {this.updateServiceAnswer} />;
}

export default ServiceAnswerDetailsContainer