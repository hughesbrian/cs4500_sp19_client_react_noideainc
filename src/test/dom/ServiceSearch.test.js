import React from 'react'
import TestRenderer from 'react-test-renderer'
import serviceAnswers from '../mockdata/ServiceAnswer2.mock.json'
import ServiceAnswerDetailsContainer from '../../components/ServiceAnswerDetailsContainer'
import sleep from '../util/sleep'

test('ProviderSearch renders correctly', async () => {
    const testRenderer = TestRenderer.create(
        <serviceAnswers/>
    )
    const testRendererq2= TestRenderer.create(
        <ServiceAnswerDetailsContainer/>
    )
    await sleep(1000) // sleep to give the component a chance to render.

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root

    const select_answer = testInstance.findAllByProps({className: 'select-answer'})
    await sleep(500) // sleep to give the component a chance to render.
    select_answer[0].props.onChange({target:{value: 1}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const select_question = testInstance.findByProps({className: 'select-question'})
    await sleep(500) // sleep to give the component a chance to render.
    select_question.props.onChange({target: {value: 1}})
    tree = testRendererq2.toJSON()
    expect(tree).toMatchSnapshot()

})
