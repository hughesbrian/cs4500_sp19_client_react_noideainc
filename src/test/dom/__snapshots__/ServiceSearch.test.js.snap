<div id="root">
    <div class="container-fluid home">
    <h1>No Idea Inc</h1>
    <div class="sum-nar-bar">
    <a href="/home">Home</a> |
    <a href="/services">
     Services
     </a> |
     <a href="/providers">
     Providers
     </a> |
     <a href="/admin">Admin
     </a> |
     <a href="/provider">
     Provider
     </a>
     <br>
     <div>
     <div class="row">
     <div class="col-8">
     <div class="input-group input-group-lg">
     <input placeholder="Search for providers" type="text" class="form-control">
     <input placeholder="Zip code" type="text" class="form-control">
     <div class="input-group-append">
     <button class="btn btn-primary" type="button">
     Search
     </button>
     </div>
     </div>
     </div>
     <div class="col-3 text-right">
     <a href="#">Sign up
     </a>
     </div>
     <div class="col-1">
     <a href="#">
     Log in
     </a>
     </div>
     </div>
     <br>

     <br>
     <ul class="nav nav-pills">
     <li class="nav-item">
     <a class="nav-link" href="#about">
     About
     </a>
     </li>
     <li class="nav-item">
     <a class="nav-link" href="#reviews">
     Reviews
     </a>
     </li>
     <li class="nav-item">
     <a class="nav-link" href="#faqs">
     FAQs
     </a>
     </li>
     </ul>
     <div class="row">
     <div class="col-2">
     <a name="about">
     </a>
     <img src="https://picsum.photos/150/150">
     </div>
     <div class="col-10">
     <h3>
     </h3>
     Number of Reviews: (1)
     </div>
     </div>
     <div>
     <br>
     <p>You want a good-looking yard? Let Nate fix it up for you!
     </p>
     </div>
     <hr>
     <div class="row">
     <div class="col-8">
     <a name="reviews">
     </a>
     <h3>
     Reviews
     </h3><
     /div>
     <hr>
     </div>
     <div class="row">
     <div class="col-4">
     1 reviews
     </div>
     </div>
     <br>
     <div>
     <div>
     <h4>
     Sam Atkins
     </h4>
     <br>
     <div>
     This place is awesome
     </div>
     <br>
     <hr>
     </div>
     <div class="center">
     <nav aria-label="...">
     <ul class="pagination">
     <li class="page-item disabled">
     <span class="page-link">
     Previous
     </span>
     </li>
     <li class="page-item">
     <a class="page-link" href="#">
     1
     </a>
     </li>
     <li class="page-item active" aria-current="page">
     <span class="page-link">
     2
     <span class="sr-only">
     (current)
     </span>
     </span>
     </li>
     <li class="page-item">
     <a class="page-link" href="#">
     3
     </a>
     </li>
     <li class="page-item">
     <a class="page-link" href="#">
     Next
     </a>
     </li>
     </ul>
     </nav>
     </div>
     </div>
     <div>
     <hr>
     <a name="faqs">
     </a>
     <h3>
     FAQs
     </h3>
     <div>
     <h5>
     Have you passed a background check?
     </h5>
     <p>
     Yes
     </p>
     <hr>
     </div>
     <div>
     <h5>
     How many employees does your company have?
     </h5>
     <p>
     500
     </p>
     <hr>
     </div>
     <div>
     <h5>
     Have you passed a background check?
     </h5>
     <p>
     no
     </p>
     <hr>
     </div>
     </div>
     </div>
     </div>
     </div>
     </div>